src/libponyc/type/cap.c

```diff
@@ -305,26 +288,40 @@ bool is_cap_sub_cap_bound(token_id sub, token_id subalias, token_id super,
       break;
 
     case TK_VAL:
-    case TK_CAP_SHARE:
       switch(super)
       {
         case TK_BOX:
-        case TK_CAP_SHARE:
+        case TK_CAP_SHARE: // {val, tag}
           return true;
 
         default: {}
       }
       break;
 
-    case TK_CAP_READ:
-    case TK_CAP_ALIAS:
+    case TK_CAP_READ: // {iso, trn, ref, val, box}
```

Comment:
`#read = {ref, val, box}`. So also a subtype of `TK_CAP_ALIAS` and `TK_CAP_SHARE` ?